include/nuttx/progmem.h

```diff
@@ -147,48 +170,50 @@ size_t up_progmem_getaddress(size_t page);
  *
  ****************************************************************************/
 
-ssize_t up_progmem_eraseblock(size_t block);
+ssize_t up_progmem_erase(size_t index);
 
 /****************************************************************************
- * Name: up_progmem_ispageerased
+ * Name: up_progmem_issectionerased
  *
  * Description:
- *   Checks whether erase page is erased
+ *   Checks whether a specified unit of memory is erased
  *
  * Input Parameters:
- *   page - The page index to be checked.
+ *   index - ordinal number of the set of erasable units of memory
  *
  * Returned Value:
  *   Returns number of bytes NOT erased or negative value on error. If it
- *   returns zero then complete page is erased.
+ *   returns zero then complete section is erased.
  *
  *   The following errors are reported:
- *     -EFAULT: On invalid page
+ *     -EFAULT: On invalid index
  *
  ****************************************************************************/
 
-ssize_t up_progmem_ispageerased(size_t page);
+ssize_t up_progmem_iserased(size_t index);
 
 /****************************************************************************
  * Name: up_progmem_write
  *
  * Description:
  *   Program data at given address
  *
- *   Note: this function is not limited to single page and nor it requires
- *   the address be aligned inside the page boundaries.
+ *   Note: This function may cross memory boundaries so long as the caller
+ *         has ensured all units of memory that will be programmed have been
+ *         erased. It also does not require that the write be aligned with the
+ *         beginning of a unit of memory.
  *
  * Input Parameters:
  *   addr  - Address with or without flash offset
- *           (absolute or aligned to page0)
+ *           (absolute or aligned to sector0)
```

Comment:
I find this confusing. 

If it is an address the caller has to know the base address of the memory, so the arch data has to be exposed to the caller.  Would could add a function that returns the the base(s) as any counted array.

If it is BASE address less it is an offset.

Reply:
@davids5 I believe I have addressed all of your comments except this one. The up_progmem_getaddress() interface could be used to get the base addresses of memory. Is this not sufficient? Or perhaps I don't understand your concern fully.

Reply:
IIRC the H7 has 2 banks of FLASH.  How would the user get the second one or know it is the second own without including arch specific code.  Maybe we should talk this one through.